### PR TITLE
chore: use pnpm to manage dependencies

### PR DESCRIPTION
**Pull request by jacobwgillespie**
## Problem

Followup from #12520 (see there for more context), this PR replaces Yarn 1 with pnpm. Yarn 1 is semi-unmaintained and has caused issues in `docker build`.

## Changes

This PR switches to use `pnpm` instead of `yarn`. The highlights are:

* All yarn commands are updated to their pnpm equivalents
* GitHub Actions workflows and the Dockerfile use pnpm now
* Some additional dependencies have been added to `package.json` if they previously were only available due to hoisting.

## How did you test this code?

Installed locally + ran `docker build`. Will need CI to run the tests.

### CODE REVIEW

**Review by Twixes — COMMENTED** [review]
_Looks_ good to me, but TypeScript is acting up, and I don't see why since nothing should be different with regard to typing. 🤔 Also something fishy going on with plugin server tests.

File: frontend/src/scenes/insights/filters/ActionFilter/ActionFilter.tsx
Comment (jacobwgillespie) [reply frontend/src/scenes/insights/filters/ActionFilter/ActionFilter.tsx:79]: That's for the ESLint rule `react/display-name` - with the arrow function inside `forwardRef`, the wrapped component is unnamed.

**Comment by jacobwgillespie** [comment]
> TypeScript is acting up, and I don't see why since nothing should be different with regard to typing

Yeah... So this is the fun of migrating from Yarn 1 to pnpm (or a more strict Yarn) - Yarn 1 hoisted all modules to the root of `node_modules`, so you could potentially import a sub-dependency in your app without specifying it in package.json, and the behavior of tools that inspect `node_modules` can change. For instance, `@types/*` packages would be hoisted to the top, ESLint plugins would be hoisted, etc.

With pnpm or a more correct Yarn version, you can only import packages specified in `package.json`. This is way safer and more predictable, but when migrating you discover all the places that previously were accidentally working due to hoisting. 🙂 

I'll keep watching the build, I think I'm getting closer to fixing all the TS issues.

File: patches/kea-forms@3.0.1.patch
Comment (Twixes) [inline patches/kea-forms@3.0.1.patch:1]: CC @mariusandra

File: frontend/src/scenes/insights/filters/ActionFilter/ActionFilter.tsx
Comment (Twixes) [reply frontend/src/scenes/insights/filters/ActionFilter/ActionFilter.tsx:79]: Oh interesting, I thought we were already compliant with this rule. Looks like it's just been a while since this file was last edited and only now did ESLint catch this.

**Review by Twixes — COMMENTED** [review]
This is looking fantastic, well done! Ran this locally (`bin/start`) and everything's working.
The only remaining issue it those few failing Cypress tests – though I ran them on a clean setup locally with fully green results, so this might be some flakiness. Also not sure why Jest tests haven't ran in CI. Once the checks are green, we can merge this.

File: patches/kea-forms@3.0.1.patch
Comment (jacobwgillespie) [reply patches/kea-forms@3.0.1.patch:1]: Ah, cool, yeah an upstream fix would be even better. 🙂 

The issue here is that `kea` in 3.0.0-alpha accepted just two generic arguments for `SelectorDefinition`, but in the 3.x.x released version requires three. Looks like `kea-forms` assumes the alpha version's types.

**Comment by jacobwgillespie** [comment]
> this might be some flakiness. Also not sure why Jest tests haven't ran in CI. Once the checks are green, we can merge this.

Yeah, having done this before, the flakiness may very well be related to some module reshuffling, and after merge / deploy it's quite possible that there might be a followup issue to fix for the release. 🙂 Whenever this is merged, it would be good to watch for any errors, etc.

File: .github/workflows/ci-frontend.yml
Comment (jacobwgillespie) [reply .github/workflows/ci-frontend.yml]: Updated 👍 

> Allow edits and access to secrets by maintainers

Apparently GitHub doesn't support this if the repo and the fork are both located in organization accounts rather than personal accounts. 🤷 https://github.com/orgs/community/discussions/5634

File: patches/kea-forms@3.0.1.patch
Comment (mariusandra) [reply patches/kea-forms@3.0.1.patch:1]: Thanks for finding this! I released version 3.0.2 of kea-forms with the fix.

File: .github/workflows/ci-frontend.yml
Comment (Twixes) [reply .github/workflows/ci-frontend.yml]: Oh, that's very annoying. Now there are some "Jest encountered an unexpected token" problems in Frontend CI checks, and unfortunately I'm not able to push a fix either.

**Review by mariusandra — COMMENTED** [review]
This looks good to me! The last changes I think we need are:
- kea-forms can now be upgraded to a version that fixes the bug
- multiple RUN commands in production.Dockerfile should be combined to one, in two places

Other than that, if there's no easy way to get CI working, we're just going to merge it in, and quickly fix things if any remaining issues arise. I think the stakes are low enough that we could get away with it.

**Review by mariusandra — APPROVED** [review]

**Comment by jacobwgillespie** [comment]
Okay, I've updated with changes from `master` and cleaned up the Dockerfile. I still need to fix Jest, which is currently not transpiling tests correctly (it still has `import` / `export`).

**Comment by mariusandra** [comment]
We recently upgraded Jest and are running it now through `esbuild`, so that problem might have solved itself.

I'd be happy to merge it in, though we decided to wait until next week as we're just about to release a new version of PostHog... and just in case it's best to wait for after that :).

**Comment by jacobwgillespie** [comment]
Closing in favor of #13190 🚀

**Comment by charlescook-ph** [comment]
Thank you so much @jacobwgillespie! Check your email for a small gift from us :)

**Comment by jacobwgillespie** [comment]
Hey, thank you! 🙌

**Pull request closed without merging**
